main.tf

```diff
@@ -392,6 +393,6 @@ resource "aws_route53_record" "master" {
   zone_id = "${data.aws_route53_zone.dns_zone.zone_id}"
   name    = "${var.cluster_name}.${var.hosted_zone}"
   type    = "A"
-  records = ["${aws_eip.master.public_ip}"]
+  records = ["${aws_instance.master.private_ip}"]
```

Comment:
This DNS record is actually supposed to be used for accessing the cluster from the outside (using kubectl from your PC etc.). So this breaks this. Moreover, it leaves Elastic IP address unused.

What I think you should do is to revert this change and either:
* Create a different hostname (e.g. something like `internal.${var.cluster_name}.${var.hosted_zone}`)
* Use the new hostname to link with the nodes
or maybe be much easier and still fully sufficient:
* Use the internal DNS record / private IP in the init_node template

That should keep the DNS record work as it worked before but also make sure that the cluster links through the private IP addresses which I agree should be much better ... even for people who have the public IP.

Reply:
That sounds a lot cleaner than this.  Let me see what i can do